src/libstore/binary-cache-store.cc

```diff
@@ -66,6 +67,24 @@ void BinaryCacheStore::init()
                 config.wantMassQuery.setDefault(value == "1");
             } else if (name == "Priority") {
                 config.priority.setDefault(std::stoi(value));
+            } else if (name == "RequiredSignatures") {
+                // Parse whitespace-separated list of public keys
+                for (const auto & keyStr : tokenizeString<Strings>(value, " \t\n\r")) {
+                    if (!keyStr.empty()) {
+                        try {
+                            PublicKey key(keyStr);
+                            requiredSignatures.emplace(key.name, key);
+                        } catch (Error & e) {
+                            e.addTrace(
+                                {},
+                                "while parsing RequiredSignatures field in binary cache '%s'",
+                                config.getHumanReadableURI());
+                            throw;
+                        }
+                    }
+                }
+            } else if (name == "RequireAllSignatures") {
```

Comment:
This could also be `NrSignaturesNeeded`, taking an arbitrary integer, similar to `nix verify`'s `--sigs-needed` flag. It might be useful to require e.g. at least 2 valid signatures.